decidim-blogs/app/models/decidim/blogs/post.rb

```diff
@@ -35,12 +36,18 @@ class Post < Blogs::ApplicationRecord
                         index_on_create: true,
                         index_on_update: ->(post) { post.visible? })
 
-      def visible?
-        participatory_space.try(:visible?) && component.try(:published?)
+      class << self
+        def all_timestamp_attributes_in_model
+          super + ["published_at"]
+        end
+
+        def log_presenter_class_for(_log)
+          Decidim::Blogs::AdminLog::PostPresenter
+        end
       end
 
-      def self.log_presenter_class_for(_log)
-        Decidim::Blogs::AdminLog::PostPresenter
+      def visible?
+        participatory_space.try(:visible?) && component.try(:published?)
```

Comment:
Maybe we should also check here that the post itself is published?

Reply:
I dont know how, and which commit you looked at, but this issue has already addressed from commit 21af73ab407d5

Reply:
I am checking the current latest revision at the `mainio:feature/publication-date-field-for-blog-articles` branch. PR reviews are always done at the latest revision at the time of the review.

You are not checking here if the post is visible, i.e. whether it's `published_at` date has already passed.

The checks being done here are:
1. Is the participatory space visible (e.g. a participatory process)
2. Is the **component** itself published

So you are not checking here if the post itself is published or not.

Reply:
Aha, I just got what you mean, I confused it with component.try(:published?)